hs_core/models.py

```diff
@@ -110,6 +110,8 @@ def get_user(request):
     :param request:
     :return: django.contrib.auth.User
     """
+    if not hasattr(request, 'user'):
```

Comment:
authorize is used all across hydroshare.  Anytime a request that doesn't have a user is made, this method would raise a WSGI request has no attribute user error.  This change will fix the problem for all requests, not just add-files-to-resource endpoint.